pandas/_libs/lib.pyx

```diff
@@ -1250,6 +1252,20 @@ def is_pyarrow_array(obj):
     return False
 
 
+def is_null_slice(obj):
+    cdef Py_ssize_t start, stop, step
+    if isinstance(obj, slice):
+        try:
+            PySlice_Unpack(obj, &start, &stop, &step)
+        except TypeError:
+            return False
+
+        if start == 0 and stop == PY_SSIZE_T_MAX and step == 1:
```

Comment:
will this behave differently from the status quo on e.g. `slice(0, None, 1)`?

Reply:
No - `PY_SSIZE_T_MAX` is the largest unit used for addressing on the platform. CPython converts `None` to that internally, essentially letting you index as far out as your platform allows

Reply:
it's not the None here that im concerned with, its the 0 and 1.  On main:

```
In [1]: import pandas.core.common as com
In [2]: obj = slice(0, None, 1)
In [3]: com.is_null_slice(obj)
Out[3]: False
```

Reply:
Hmm nice catch. That does return `True` here - are you thinking that is better or worse?

Reply:
i think the function is intended pretty specifically to catch `obj[:]`, so would keep the status quo behavior

Reply:
OK cool. Thanks for catching that - I think can just close this for now then. Not a super high priority